diff --git a/uniconfig/python/frinx_api/uniconfig/device/discovery/discover.py b/uniconfig/python/frinx_api/uniconfig/device/discovery/discover.py
index b6acba2..c1be7fc 100644
--- a/uniconfig/python/frinx_api/uniconfig/device/discovery/discover.py
+++ b/uniconfig/python/frinx_api/uniconfig/device/discovery/discover.py
@@ -79,7 +79,7 @@ class TypeOfAddressModel3(BaseModel):
     """
 
 
-class Addres(BaseModel):
+class Address(BaseModel):
     model_config = ConfigDict(
         populate_by_name=True,
     )
@@ -129,7 +129,7 @@ class Input(BaseModel):
     Check whether the host is reachable or not using ICMP protocol
     """
     tcp_port: Optional[list[TcpPortItem]] = Field(None, alias='tcp-port')
-    address: Optional[list[Addres]] = None
+    address: Optional[list[Address]] = None
     udp_port: Optional[list[UdpPortItem]] = Field(None, alias='udp-port')
 
 
