diff --git a/uniconfig/python/frinx_api/uniconfig/device/discovery/discover.py b/uniconfig/python/frinx_api/uniconfig/device/discovery/discover.py
index 86325f9..8e90bf6 100644
--- a/uniconfig/python/frinx_api/uniconfig/device/discovery/discover.py
+++ b/uniconfig/python/frinx_api/uniconfig/device/discovery/discover.py
@@ -118,6 +118,34 @@ class UdpPortItem(BaseModel):
     ] = Field(None, alias='device-discovery:type-of-port')
 
 
+class AvailableUdpPort(RootModel[int]):
+    model_config = ConfigDict(
+        populate_by_name=True,
+    )
+    root: int = Field(..., ge=0, le=65535)
+
+
+class UnavailableTcpPort(RootModel[int]):
+    model_config = ConfigDict(
+        populate_by_name=True,
+    )
+    root: int = Field(..., ge=0, le=65535)
+
+
+class AvailableTcpPort(RootModel[int]):
+    model_config = ConfigDict(
+        populate_by_name=True,
+    )
+    root: int = Field(..., ge=0, le=65535)
+
+
+class UnavailableUdpPort(RootModel[int]):
+    model_config = ConfigDict(
+        populate_by_name=True,
+    )
+    root: int = Field(..., ge=0, le=65535)
+
+
 class Input(BaseModel):
     model_config = ConfigDict(
         populate_by_name=True,
@@ -141,8 +169,8 @@ class DeviceItem(BaseModel):
     """
     Host address either in IP (IPv4 or IPv6) format or in domain-name format
     """
-    unavailable_tcp_ports: Optional[list[int]] = Field(
-        None, alias='unavailable-tcp-ports', ge=0, le=65535
+    unavailable_tcp_ports: Optional[list[UnavailableTcpPort]] = Field(
+        None, alias='unavailable-tcp-ports'
     )
     """
     TCP ports that are unreachable
@@ -151,20 +179,20 @@ class DeviceItem(BaseModel):
     """
     If the host is reachable or not using ICMP protocol
     """
-    available_tcp_ports: Optional[list[int]] = Field(
-        None, alias='available-tcp-ports', ge=0, le=65535
+    available_tcp_ports: Optional[list[AvailableTcpPort]] = Field(
+        None, alias='available-tcp-ports'
     )
     """
     All the available TCP ports
     """
-    unavailable_udp_ports: Optional[list[int]] = Field(
-        None, alias='unavailable-udp-ports', ge=0, le=65535
+    unavailable_udp_ports: Optional[list[UnavailableUdpPort]] = Field(
+        None, alias='unavailable-udp-ports'
     )
     """
     UDP ports that are unreachable
     """
-    available_udp_ports: Optional[list[int]] = Field(
-        None, alias='available-udp-ports', ge=0, le=65535
+    available_udp_ports: Optional[list[AvailableUdpPort]] = Field(
+        None, alias='available-udp-ports'
     )
     """
     All the available UDP ports
     """
